cookbooks/fb_networkmanager/libraries/default.rb

```diff
@@ -0,0 +1,57 @@
+#
+# Cookbook:: fb_networkmanager
+# Recipe:: default
+#
+# Copyright (c) 2020-present, Vicarious, Inc.
+# Copyright (c) 2020-present, Facebook, Inc.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'iniparse'
+
+module FB
+  class Networkmanager
+    def self.active_connections
+      return {} unless ::File.exist?('/usr/bin/nmcli')
+
+      s = Mixlib::ShellOut.new('nmcli -t conn show --active').run_command
+      return {} if s.error?
+
+      cons = {}
+      s.stdout.each_line do |line|
+        name, uuid, type, device = line.strip.split(':')
+        cons[name] = {
+          'uuid' => uuid,
+          'type' => type,
+          'device' => device,
+        }
+      end
+      cons
+    end
+
+    def self.to_ini(content)
+      IniParse.gen do |doc|
+        content.each_pair do |sect, opts|
+          doc.section(sect, :option_sep => '=') do |section|
+            opts.each_pair do |opt, val|
+              v = [val].flatten.join(',')
```

Comment:
Can you give me an example to test? I had `eap=peap;` and dropped it to `eap=peap` and it worked fine.

Reply:
@jgoguen I don't think the trailing slash is necessary, and here's some data:

```
[phil@ldx-vulturus ~]$ sudo grep ^eap /etc/NetworkManager/system-connections/fb_networkmanager_vicarious_wifi_hq
eap=peap
[phil@ldx-vulturus ~]$ nmcli c show 6f1f4851-15a3-321f-beec-fbfc873c9f85 | grep 802-1x.eap
802-1x.eap:                             peap
```

vs

```
root@ldx-vulturus:~# sudo grep ^eap /etc/NetworkManager/system-connections/fb_networkmanager_vicarious_wifi_hq
eap=peap;
root@ldx-vulturus:~# nmcli c show 6f1f4851-15a3-321f-beec-fbfc873c9f85 | grep 802-1x.eap
802-1x.eap:                             peap
```

They're identical.

Reply:
Interesting, what version of NetworkManager? I have 1.20 and even for connections manually created fresh it's still forcing semicolon-separate lists with a trailing semicolon.

I have 1.22 on another laptop, I can check what it does later tonight.

Reply:
1.10.6

Reply:
I wonder then if this INI-incompatible change is the new format. I created new connections on both 1.20 and 1.22, both forced trailing semicolons for multi-value fiends. It would work if I remove the trailing semicolon, if I restart NetworkManager itself not just reload connections, but as soon as anything modified the connection file it put the trailing semicolons back.

Where it uses semicolons as list item separators, I wonder if `IniParse` would see the first value and then discard the rest treating it as a comment…

Reply:
certainly it does like to add them back. I just didn't find any problem to not having them.

Reply:
Well, I'm going to assume there was something borked on my system. So as long as this can output semicolon-delimited lists for the connection files and comma-delimited lists for `NetworkManager.conf` it's good.

Reply:
It doesn't today. I'll add that and tests. Thanks!

Reply:
oh huh, I didn't get to this... I'll come back to this